src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -1400,138 +1400,150 @@ void TaskSketcherElements::updateVisibility()
                 }
-                // update the listwidget
-                {
-                    QSignalBlocker sigblk(ui->listWidgetElements);
-                    if (modified_item != NULL) {
-                        bool is_selected = modified_item->isSelected();
-                        const bool should_be_selected = modified_item->isLineSelected
-                            || modified_item->isStartingPointSelected || modified_item->isEndPointSelected
-                            || modified_item->isMidPointSelected;
-
-                        // If an element is already selected and a new subelement gets selected
-                        // (eg., if you select the arc of a circle then select the center as well),
-                        // the new subelement won't get highlighted in the list until you mouseover
-                        // the list.  To avoid this, we deselect first to trigger a redraw.
-                        if (should_be_selected && is_selected) {
-                            modified_item->setSelected(false);
-                            is_selected = false;
-                        }
-
-                        if (should_be_selected != is_selected) {
-                          modified_item->setSelected(should_be_selected);
+            }
+            else if (boost::starts_with(shapetype, "Vertex")) {
+                QRegularExpression rx(QStringLiteral("^Vertex(\\d+)$"));
+                QRegularExpressionMatch match;
+                boost::ignore_unused(expr.indexOf(rx, 0, &match));
+                if (!match.hasMatch()) {
+                    return;
+                }
+                bool ok;
+                int ElementId = match.captured(1).toInt(&ok) - 1;
+                if (!ok) {
+                    return;
+                }
+                // Get the GeoID&Pos
+                int GeoId;
+                Sketcher::PointPos PosId;
+                sketchView->getSketchObject()->getGeoVertexIndex(ElementId, GeoId, PosId);
+
+                int countItems = ui->listWidgetElements->count();
+                for (int i = 0; i < countItems; i++) {
+                    ElementItem* item = static_cast<ElementItem*>(ui->listWidgetElements->item(i));
+                    if (item->ElementNbr == GeoId) {
+                        modified_item = item;
+                        switch (PosId) {
+                            case Sketcher::PointPos::start:
+                                item->isStartingPointSelected = select;
+                                break;
+                            case Sketcher::PointPos::end:
+                                item->isEndPointSelected = select;
+                                break;
+                            case Sketcher::PointPos::mid:
+                                item->isMidPointSelected = select;
+                                break;
+                            default:
+                                break;
                         }

```

Comment:
Was there a function for this somewhere? Feels like a common operation (set correct point in item)

Reply:
Doesn't seem so. IIRC I was looking at making one, but probably didn't for whatever reason.